### ISSUE TITLE
CTD after launching from planet on Apple Silicon after e0153c2

### ISSUE THREAD

**Issue by bene-dictator** [endless-sky/endless-sky#9674]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When launching from a planet, Endless Sky crashes. Occurs when built from source and from ESL2. Appears to only happen after commit e0153c2. Corroborated by Alex Rov on Discord.

VSCode log says it is a segmentation fault.

### Steps to Reproduce

1. Create a new pilot.
2. Buy a ship.
3. Ignore James.
4. Launch.

### Expected Behavior

No CTD, ship launches.

### Operating System

2020 MacBook Pro (M1), macOS Sonoma 14.1.2

### Game Source

Built from source

### Game Version

4ee3a4f

**Comment by alexrovw** [comment]
Confirmed, I get 100% crash when departing a planet after checking out this commit. I am also running on Apple Silicon.
```
commit e0153c2cc66f1b7d00fe75b143b45eee9dbcf67d (HEAD)
Author: TomGoodIdea <108272452+TomGoodIdea@users.noreply.github.com>
Date:   Mon Jan 15 20:32:11 2024 +0100

    feat(enhancement): Added a message log panel for reviewing past hails and status messages (#9246)

**Comment by RisingLeaf** [comment]
Found a possible logic error, please test

**Comment by alexrovw** [comment]
Fixed in 216fa14b6a10bc9b634f3027b78c284c884f6c0e

Thanks!

**Comment by quyykk** [comment]
Fixed in #9675 216fa14b6a10bc9b634f3027b78c284c884f6c0e
